src/mappings.py

```diff
@@ -0,0 +1,213 @@
+        "Source": sources,
+        "Reference": references,
+        "DisorderMappingRelation": mapping_relations,
+        "DisorderMappingValidationStatus": validation_statuses,
+        "DisorderMappingICDRelation": icd_relations,
+        "DisorderMappingICDRefUri": icd_ref_uris
+    }
+    df = pd.DataFrame(data)
+    return df
+
+
+def ordo_sssom(
+    input_nomenclature_xml: str = INPATH_NOMENCLATURE_XML, input_sssom_config: str = INPUT_CONFIG,
+    outpath_ordo_mappings: str = OUTPATH_ORDO,
+) -> pd.DataFrame:
+    """Create ORDO SSSOM mappings"""
+    # Get data
+    df: pd.DataFrame = xml_as_df(input_nomenclature_xml)
+
+    # Create / rename columns
+    df['predicate_id'] = df['DisorderMappingRelation'].apply(
+        lambda x: 'skos:exactMatch' if x.startswith('E (')
+        else 'skos:narrowMatch' if x.startswith('NTBT (')
+        else 'skos:broadMatch' if x.startswith('BTNT (')
+        else 'skos:relatedMatch')  # 0 instances of relatedMatch; just a fallback
+    df['subject_id'] = df['OrphaCode'].apply(lambda x: 'Orphanet:' + str(x))
+    df['object_id'] = df['DisorderMappingICDRefUri'].apply(lambda x: 'icd11.foundation:' + str(x) if x else None)
+    df = df.rename(columns={'Name': 'subject_label'})
+
+    # Filter by only if ICD11Foundation ref exists && is exact match
+    df = df[(df['DisorderMappingICDRefUri'].notna()) & (df['predicate_id'] == 'skos:exactMatch')]
+
+    # Add mapping_justification
+    df['mapping_justification'] = 'semapv:UnspecifiedMatching'
+
+    # Add labels
+    lab_df = pd.read_csv(INPUT_ICD11_LABELS, sep='\t').rename(columns={'id': 'object_id', 'label': 'object_label'})
+    df = pd.merge(df, lab_df, on='object_id', how='left')
+
+    # Get only columns we care about
+    df = df[['subject_id', 'subject_label', 'predicate_id', 'object_id', 'object_label', 'mapping_justification']]
+
+    # Save & return
+    write_sssom(df, input_sssom_config, outpath_ordo_mappings)
+    return df
+
+
+def mondo_sssom(
+    df: pd.DataFrame, input_mondo_sssom: str = INPUT_MONDO_SSSOM, input_sssom_config: str = INPUT_CONFIG,
+    outpath_mondo_mappings: str = OUTPATH_MONDO,
+) -> pd.DataFrame:
+    """Create Mondo SSSOM mappings"""
+    mondo_sssom_df = pd.read_csv(input_mondo_sssom, sep='\t', comment='#')
+    mondo_sssom_df['object_prefix'] = mondo_sssom_df['object_id'].apply(lambda x: x.split(':')[0])
+    mondo_sssom_df = mondo_sssom_df[
+        (mondo_sssom_df['predicate_id'] == 'skos:exactMatch') &
+        (mondo_sssom_df['object_prefix'] == 'Orphanet')
+    ].rename(
+        columns={'object_id': 'ordo_id'}
+    )[['subject_id', 'subject_label', 'ordo_id']]
+
+    df2 = (df.rename(columns={'subject_id': 'ordo_id', 'subject_label': 'ordo_label'})\
+        .merge(mondo_sssom_df, how='inner', on='ordo_id')[[
+            'subject_id', 'subject_label', 'predicate_id', 'object_id', 'object_label', 'ordo_id',
+            'mapping_justification']]
+        .rename(columns={'ordo_id': 'match_string'}))
```

Comment:
## `match_string`
- [ ] @matentzn review

@matentzn asked for this [on Slack](https://monarchinitiative.slack.com/archives/C03AW548J8M/p1713300259182229?thread_ts=1713225589.907679&cid=C03AW548J8M) and I completed and reported there. But I wasn't sure if I did it correctly.

As asked, I added the ORDO IDs that were the intermediary for the Mondo<->ICD mappings.

I wrote:
> Maybe it’s good as-is. But if you want it formalized, I guess I could add a `subject_match_field` with values `skos:exactMatch` or something.
> I’m looking at this file as a reference for `match_string`, but this is how it’s used in lexmatch, so I guess that’s different from our case here

- [Link to `match_string` docs](https://mapping-commons.github.io/sssom/match_string/)


<details><summary>Example use of match_string in mondo-ingest</summary>
<p>


|subject_id|subject_label|object_id|predicate_id|object_label|mapping_justification|mapping_tool|confidence|subject_match_field|object_match_field|match_string|comment|predicate_modifier|reviewer_id|rejection_reason|
|-|-|-|-|-|-|-|-|-|-|-|-|-|-|-|
|MONDO:0005571|polycythemia|NCIT:C27794|skos:exactmatch|Polycythemia (Excluding Polycythemia Vera)|semapv:LexicalMatching|oaklib|0.8497788952|rdfs:label|rdfs:label|polycythemia|LEXMATCH|Not|https://orcid.org/0000-0002-4142-7153|mapping rule incorrect|
|MONDO:0016642|meningioma|DOID:0080842|skos:exactmatch|intracranial meningioma|semapv:LexicalMatching|oaklib|0.8|oio:hasExactSynonym|rdfs:label|intracranial meningioma|LEXMATCH|Not|https://orcid.org/0000-0002-4142-7153|synonym type incorrect|
|MONDO:0016715|ependymoblastoma|DOID:0081286|skos:exactmatch|embryonal tumor with multilayered rosettes|semapv:LexicalMatching|oaklib|0.8|oio:hasExactSynonym|rdfs:label|embryonal tumor with multilayered rosettes|LEXMATCH|Not|https://orcid.org/0000-0002-4142-7153|split/merge difference in Mondo vs source|

</p>
</details>

Reply:
The better way in SSSOM would be to model how we do "chained mappings", e.g. a->b and b->c, therefore a->c.

So since that hasn't been formalized yet, we're using this `match_string` field in an unconventional way to represent that.